### ISSUE TITLE
How to build the latest version on Ubuntu 16.04

### ISSUE THREAD

**Issue by MrOCR** [Debian/debhelper#2]
How can I build the latest version on Ubuntu 16.04?

**Comment by nthykier** [comment]
Mr. Ocr:
> How can I build the latest version on Ubuntu 16.04?
> 

Hi,

There are some people in Ubuntu, who seem to be working on this (see
https://bugs.launchpad.net/ubuntu/+source/debhelper/+bug/1728423).  I
believe their work will be a good starting point (and simply need to be
rebased on the newest version of debhelper).

The issue you will run into is that you will either need to backport
some other packages (e.g. dh-autoreconf) or reduce debhelpers
dependencies/breaks on them.  The latter having the consequence that
some things will not work as intended out of the box.  I have tried to
document these issues in `debian/control` next to the relation.

Finally, you will have to apply Ubuntu's changes to debhelper as well.
The code changes here are relatively modest; but I am unsure how you
best handle the dbgsym packages.  In 16.04, Ubuntu has some "shadow"
tooling on their build systems to auto-generate them while in the newest
versions of Ubuntu, they use the dbgsyms made by debhelper.  I have no
clue which is the right choice for a backport to 16.04 (but hopefully
the Ubuntu bug I linked can help there).

I think this is about all I can say in general about making debhelper
work on Ubuntu 16.04.  If you run into concrete issues, I am happy to
help answering those but I am very rusty on the Ubuntu specific things.

Hope the above helps; otherwise feel free to ask follow up questions.
